src/NzbDrone.Core/Download/Clients/Porla/Porla.cs

```diff
@@ -0,0 +1,281 @@
+                    continue;
+                }
+
+                var outputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(torrent.SavePath));
+
+                var item = new DownloadClientItem
+                {
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this, false),
+                    DownloadId = torrent.InfoHash.Hash,
+                    OutputPath = outputPath + torrent.Name,
+                    RemainingSize = torrent.Total,
+                    RemainingTime = torrent.ETA < 1 ? (TimeSpan?)null : TimeSpan.FromSeconds((double)torrent.ETA), // LibTorent uses `-1` to denote "infinite" time i.e. I am stuck, or I am done. FromSeconds will convert `-1` to 1, so we need to do this.
+                    Title = torrent.Name,
+                    TotalSize = torrent.Size,
+                    SeedRatio = torrent.Ratio
+                };
+
+                // deal with moving_storage=true ?
+
+                if (!string.IsNullOrEmpty(torrent.Error))
+                {
+                    item.Status = DownloadItemStatus.Warning;
+                    item.Message = torrent.Error;
+                } // TODO: check paused or finished first?
+                else if (torrent.FinishedDuration > 0)
+                {
+                    item.Status = DownloadItemStatus.Completed;
+                    if (torrent.ETA < 1)
+                    {
+                        // Sonarr wants to see a TimeSpan.Zero when it is done ( -1 -> 0 )
+                        item.RemainingTime = TimeSpan.Zero;
+                    }
+                }
+                else if (torrent.Flags.Contains("paused"))
+                {
+                   item.Status = DownloadItemStatus.Paused;
+                } /*  I don't know what the torent looks like if it is "Queued"
+                else if (???)
+                {
+                   item.Status = DownloadItemStatus.Queued;
+                } */
+                else
+                {
+                    item.Status = DownloadItemStatus.Downloading;
+                }
+
+                item.CanMoveFiles = item.CanBeRemoved = true; // usure of what restricts this on porla. Currently these is always true
+
+                items.Add(item);
+            }
+
+            if (items.Count < 1)
+            {
+                _logger.Debug("No Items Returned");
+            }
+
+            return items;
+        }
+
+        public override void RemoveItem(DownloadClientItem item, bool deleteData)
+        {
+            // Kinda sucks we don't have a `RemoveItems`, porla has a batch interface for removals
+            PorlaTorrent[] singleItem = { new PorlaTorrent(item.DownloadId, "") };
+            _proxy.RemoveTorrent(Settings, deleteData, singleItem);
+
+            // when do we set item.Removed ?
```

Comment:
No need to, when it's dropped from the client it'll be removed from Sonarr's queue.